clients/unshaded/src/main/java/tachyon/client/file/AbstractTachyonFileSystem.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.file;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+import tachyon.TachyonURI;
+import tachyon.annotation.PublicApi;
+import tachyon.client.FileSystemMasterClient;
+import tachyon.thrift.BlockInfoException;
+import tachyon.thrift.FileAlreadyExistException;
+import tachyon.thrift.FileDoesNotExistException;
+import tachyon.thrift.FileInfo;
+import tachyon.thrift.InvalidPathException;
+
+/**
+ * Tachyon File System client. This class should be used to interface with the Tachyon File
+ * System master and supports all non IO operations. Implementing classes should provide their
+ * own IO methods separate from the {@link TachyonFileSystemCore} interface.
+ */
+@PublicApi
+public abstract class AbstractTachyonFileSystem implements TachyonFileSystemCore {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  /** The file system context which contains shared resources, such as the fs master client */
+  // TODO (calvin): Make this private after StreamingTachyonFileSystem is cleaned up
+  protected FileSystemContext mContext;
+
+  /**
+   * Constructor, currently TachyonFileSystem does not retain any state
+   */
+  protected AbstractTachyonFileSystem() {
+    mContext = FileSystemContext.INSTANCE;
+  }
+
+  /**
+   * {@inheritDoc}
```

Comment:
Question on convention: do you need it? I think the override annotation might be enough.